pkg/controller/logcollector/logcollector_controller.go

```diff
@@ -226,7 +286,10 @@ func (r *ReconcileLogCollector) Reconcile(ctx context.Context, request reconcile
 		r.status.SetDegraded("Error querying license", err.Error())
 		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
 	}
-
+	isModified, modifiedFields, err := r.fillDefaults(ctx, instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
```

Comment:
Please move patching the instance to after this so we can Patch as soon as possible.